sync/syncer.go

```diff
@@ -115,6 +139,11 @@ func (s *Syncer[H]) Start(ctx context.Context) error {
 	}
 	// start syncLoop only if Start is errorless
 	go s.syncLoop()
+	select {
+	case <-s.started:
```

Comment:
If I understood correctly, this case will never happen. Why not just close the channel or write smth to it, so it will unblock subjective initialisation?

Reply:
There was a time when there were fraud proofs. Now all the components are stopped and started once. But it doesn't hurt to have it for convenience in tests

Reply:
> restart with a Stop followed by a Start. 

In this case, we'll get panic with the current approach:
1) closing channel for the first time will unblock read in `setVerifier`.
2) on the next attempt to close it - it will panic with `close of closed channel`.
So, we need to either recreate a channel in start or make it local to Start() method

Reply:
> There is a also a test confirming this

The test that you are referring to is using DummySubscriber with an empty verifier. It does not emulate the behaviour similar to the real network but ok.